### PR TITLE
patch CVE-2016-7103 regarding jquery dialog

### PR DESCRIPTION
**Pull request by C0rby**
## Description

 Patched a CVE regarding the jquery dialog widget. This widget is currently not being used so this is a preventive patch.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Technical debt

## Checklist:

- [x] Code changes
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by JammingBen** [comment]
@C0rby Thx for providing the fix. I assume this is relevant for 10.9, therefore please change the destination to the `release-10.9.0` branch.

**Review by AlexAndBear — APPROVED** [review]

**Comment by JammingBen** [comment]
I'm afraid just changing the destination branch is not sufficient, now we have several commits which belong to master, but not in 10.9 🙈

**Comment by C0rby** [comment]
Right, yeah let me rebase.
I also need to fix the patch because of the failing test.

**Comment by C0rby** [comment]
Now the CI should become green.

File: core/js/js.js [hunk 1]
```diff
@@ -3067,4 +3067,66 @@ $.datepicker._attachments = function (input, inst) {
...
+			}
+		}
+	} );
+	// Support: IE
+	// Use type="button" to prevent enter keypresses in textboxes from closing the
+	// dialog in IE (#9312)
+	this.uiDialogTitlebarClose = $( "<button type='button'></button>" )
+		.button( {
+			label: this.options.closeText,
+			label: $( "<a>" ).text( this.options.closeText ).html(),
```
Comment (JammingBen) [inline core/js/js.js:3113]: Which `label` should be used here?

**Review by JammingBen — COMMENTED** [review]
LGTM overall, minor nitpick because of the duplicated `label` property. SonarCloud will probably still fail because of missing test coverage, Michael can merge it though.

File: core/js/js.js (hunk 1 above)
Comment (C0rby) [reply core/js/js.js:3113]: Good catch...

**Review by JammingBen — APPROVED** [review]

**Comment by phil-davis** [comment]
> SonarCloud will probably still fail because of missing test coverage, Michael can merge it though.

It is not listed as a required passing result. So any of us can merge this now - it has been approved.

**Comment by JammingBen** [comment]
Two approvals, somewhat time-critical -> let's merge it 👍

**Pull request merged**